File: spl-net/src/main/java/bgu/spl171/net/impl/TFTP/Commands/DeleteRequest.java
package bgu.spl171.net.impl.TFTP.Commands;

import bgu.spl171.net.impl.TFTP.Commands.Responses.Response;

public class DeleteRequest implements Command {
    String fileName;

    public String getFileName() {
        return fileName;
    }

    public DeleteRequest(String fileName) {
        this.fileName = fileName;
    }

    @Override
    public short getOpCode() {
        return OpCode.Delete.getValue();
    }

    @Override
    public Response execute() {
        return null;
    }
}

File: java to c++/DeleteRequest.cpp
#include "DeleteRequest.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace Commands
					{
						using Response = bgu::spl171::net::impl::TFTP::Commands::Responses::Response;

						std::wstring DeleteRequest::getFileName()
						{
							return fileName;
						}

						DeleteRequest::DeleteRequest(const std::wstring &fileName)
						{
							this->fileName = fileName;
						}

						short DeleteRequest::getOpCode()
						{
							return OpCode::Delete::getValue();
						}
					}
				}
			}
		}
	}
}
